can/broadcastmanager.py

```diff
@@ -154,16 +148,17 @@ def _check_modified_messages(self, messages: Tuple[Message, ...]) -> None:
 
         :raises ValueError: If the given messages are invalid
         """
-        if len(self.messages) != len(messages):
+        if self.msgs_len != len(messages):
             raise ValueError(
                 "The number of new cyclic messages to be sent must be equal to "
                 "the number of messages originally specified for this task"
             )
-        if self.arbitration_id != messages[0].arbitration_id:
-            raise ValueError(
-                "The arbitration ID of new cyclic messages cannot be changed "
-                "from when the task was created"
-            )
+        for idx in range(self.msgs_len):
+            if self.arbitration_id[idx] != messages[idx].arbitration_id:
+                raise ValueError(
+                    "The arbitration ID of new cyclic messages cannot be changed "
+                    "from when the task was created"
+                )
```

Comment:
I haven't tried but do you know if socketcan supports reordering the messages as well? In which case prefer to count the arbitration ids match.